azavea/raster-vision
# Add more notebooks + misc. doc fixes and improvements

## Overview

Brief description of what this PR does, and why it is needed.

Notebook updates/addditions:
- Added a brief discussion of the `Labels` class to [reading_label.ipynb](https://raster-vision--1506.org.readthedocs.build/en/1506/tutorials/reading_labels.html#Labels)
- Added new notebook: [scenes_and_aois.ipynb](https://raster-vision--1506.org.readthedocs.build/en/1506/tutorials/scenes_and_aois.html)
- Added new notebook: [train.ipynb](https://raster-vision--1506.org.readthedocs.build/en/1506/tutorials/train.html)
- Added new notebook: [pred_and_eval.ipynb](https://raster-vision--1506.org.readthedocs.build/en/1506/tutorials/pred_and_eval_ss.html)
- ~Added an open-in-Colab button and installation instructions to all notebooks~ (See discussion in comments below).

Other changes:
- Problem with reference generation for `rastervision.pipeline.registry` and `rastervision.pipeline.rv_config` has been fixed by renaming the global `registry` and `rv_config` variables to `registry_` and `rv_config_` respectively.
- `rastervision2` plugin aliases have been removed.
- Fixed a problem with `LearnerPipelineConfig` not getting registered when `rastervision.pytorch_leaner` is the first RV plugin imported.
- Slightly refactored `Leaner.from_model_bundle`.
- Uploaded `LearnerPipelineConfig` bundles (in addition to the full pipeline bundles) to v0.13 model zoo.
    - Updated the `test.py` script to enable this.
- Made `test_ds` an optional input to `Learner`.
- Bumped `tensorboard` version.
- Added tests for `from_model_bundle` to the `Learner` unit tests.
- [One of the examples in reading_labels.ipynb](https://raster-vision--1506.org.readthedocs.build/en/1506/tutorials/reading_labels.html#Semantic-Segmentation---SemanticSegmentationLabelSource) used a non-public S3 URI to a Potsdam image before. Uploaded (only) this image to the public repo and updated the link.
- Enabled the use of `jinja` templates in all `.rst` files.
- Removed version hardcoding from docs.
- Updated version to `0.20-dev`.
- Misc. minor docs-related changes.

### Checklist


### Notes

N/A

## Testing Instructions

See https://raster-vision--1506.org.readthedocs.build/en/1506.

Makes progress on #1459 
Makes progress on #1505
